sdk/java/pulumi/src/main/java/com/pulumi/deployment/internal/DeploymentTests.java

```diff
@@ -296,48 +304,26 @@ private void initUnset() {
 
             if (this.state == null) {
                 this.state = new DeploymentImpl.DeploymentState(
-                        config,
+                        this.config,
                         this.standardLogger,
-                        options.getProjectName(),
-                        options.getStackName(),
-                        options.isPreview(),
-                        engine,
-                        monitor
+                        this.options.getProjectName(),
+                        this.options.getStackName(),
+                        this.options.isPreview(),
+                        this.engine,
+                        this.monitor
                 );
             }
-        }
-
-        public DeploymentMock setSpyGlobalInstance() {
-            initUnset();
-
-            this.deployment = Mockito.spy(new DeploymentImpl(this.state));
-            this.runner = this.deployment.getRunner();
 
-            DeploymentImpl.setInstance(new DeploymentInstanceInternal(this.deployment));
-            return new DeploymentMock(options, runner, engine, monitor, deployment, config, state, standardLogger, logger, log);
-        }
-
-        public DeploymentMock setMockGlobalInstance() {
-            initUnset();
-
-            var mock = Mockito.mock(DeploymentImpl.class);
-            //noinspection ConstantConditions
-            Mockito.when(mock.isDryRun()).thenReturn(this.state.isDryRun);
-            Mockito.when(mock.getProjectName()).thenReturn(this.state.projectName);
-            Mockito.when(mock.getStackName()).thenReturn(this.state.stackName);
-            Mockito.when(mock.getConfig(ArgumentMatchers.anyString())).then(invocation ->
-                    this.state.config.getConfig((String) invocation.getArguments()[0])
-            );
-            Mockito.when(mock.isConfigSecret(ArgumentMatchers.anyString())).then(invocation ->
-                    this.state.config.isConfigSecret((String) invocation.getArguments()[0])
-            );
-            Mockito.when(mock.getRunner()).thenReturn(this.runner);
-
-            this.deployment = mock;
+            if (this.deploymentFactory == null) {
+                this.deploymentFactory = DeploymentImpl::new;
```

Comment:
I've made real runner and real deployment the default, this is what a user would like to use, we might want to use a MockDeployment internally but this feels like the right default